packages/nbdime/src/common/mergeview.ts

```diff
@@ -143,563 +341,404 @@ function createNbdimeMergeView(
+          sources.push(r.source);
+        }
+        let decorationKey = this.getDecorationKey(sources);
+        let startingOffset = posToOffset(editor.state.doc, {
+          line: r.from.line,
+          column: r.from.column
+        });
+        let endingOffset = posToOffset(editor.state.doc, {
+          line: r.to.line,
+          column: r.to.column
+        });
+        effects.push(addHighlightEffect.of({
+          from: startingOffset,
+          to: endingOffset,
+          highlightType: highlightType,
+          decorationKey: decorationKey
+        }));
       }
     }
+    return effects;
   }
-}
 
-function highlightChars(editor: CodeMirror.Editor, ranges: DiffRangePos[],
-                        markers: Marker[], cls: string) {
-  let doc = editor.getDoc();
-  let origCls: string | null = null;
-  if (valueIn(cls, [mergeClassPrefix.del, mergeClassPrefix.insert])) {
-    origCls = cls;
-  }
-  for (let r of ranges) {
-    if (origCls !== null) {
-      cls = origCls + (r.source ? '-' + r.source.action : '');
+  private clearCharacterHighlighting(
+    editor: EditorView,
+    diffRanges: DiffRangePos[],
+    markType: DIFF_OP
+  ) {
+    let effects: StateEffect<unknown>[] = [];
+    let sources: ChunkSource[] = [];
+    if (markType === DIFF_OP.DIFF_INSERT || markType === DIFF_OP.DIFF_DELETE) {
+      let highlightType: string = markType === DIFF_OP.DIFF_DELETE ? 'deleted' : 'inserted';
+      for (let r of diffRanges) {
+        if (r.source !== undefined) {
+          sources.push(r.source);
+        }
+        let decorationKey = this.getDecorationKey(sources);
+        effects.push(removeHighlightEffect.of({highlightType: highlightType, decorationKey: decorationKey}));
+      }
     }
-    markers.push(doc.markText(r.from, r.to, {className: cls}));
+    return effects;
   }
-}
-
 
-// Updating the gap between editor and original
+  protected updateHighlighting(
+    editor: EditorView,
+    diffRanges: DiffRangePos[],
+    chunkArray: Chunk[],
+    type: DIFF_OP
+  ) {
+    let self = this;
+    let LineHighlightEffects: StateEffect<unknown>[] =
+      self.buildLineEffects(editor, chunkArray);
+    let MarkHighlightEffects: StateEffect<unknown>[] =
+      self.buildCharacterHighlighting(editor, diffRanges, type);
+    let effects: StateEffect<unknown>[] =
+      LineHighlightEffects.concat(MarkHighlightEffects);
+    editor.dispatch({ effects });
+  }
 
+  protected clearHighlighting(
+    editor: EditorView,
+    diffRanges: DiffRangePos[],
+    chunkArray: Chunk[],
+    type: DIFF_OP
+  ) {
+    let self = this;
+    let clearLineEffects: StateEffect<unknown>[] =
+      self.clearLineEffects(editor, chunkArray);
+    let clearCharacterHighlightEffects: StateEffect<unknown>[] =
+      self.clearCharacterHighlighting(editor, diffRanges, type);
+    let effects: StateEffect<unknown>[] =
+      clearLineEffects.concat(clearCharacterHighlightEffects);
+    editor.dispatch({ effects });
```

Comment:
Same remarks here:
- variables should be const
- no need for `self=this`
- the method should be private

Reply:
This has been done.